sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1448,6 +1466,64 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
                 .map(resp -> toResourceResponse(resp, Document.class));} );
     }
 
+    @Override
+    public Mono<ResourceResponse<Document>> patchDocument(String documentLink,
+                                                          List<PatchOperation<?>> patchOperations,
+                                                          RequestOptions options) {
+        DocumentClientRetryPolicy requestRetryPolicy = this.resetSessionTokenRetryPolicy.getRequestPolicy();
+        if (options == null || options.getPartitionKey() == null) {
+            String collectionLink = Utils.getCollectionName(documentLink);
+            requestRetryPolicy = new PartitionKeyMismatchRetryPolicy(collectionCache, requestRetryPolicy, collectionLink, options);
+        }
+
+        DocumentClientRetryPolicy finalRetryPolicyInstance = requestRetryPolicy;
+        return ObservableHelper.inlineIfPossibleAsObs(() -> patchDocumentInternal(documentLink, patchOperations, options, finalRetryPolicyInstance), requestRetryPolicy);
+    }
+
+    private Mono<ResourceResponse<Document>> patchDocumentInternal(String documentLink,
+                                                                   List<PatchOperation<?>> patchOperations,
+                                                                   RequestOptions options,
+                                                                   DocumentClientRetryPolicy retryPolicyInstance) {
+        if (patchOperations == null) {
+            throw new IllegalArgumentException("patchOperations");
+        }
+
+        logger.debug("Running patch operations on document. patch Operations: [{}]", patchOperations);
+
+        final String path = Utils.joinPath(documentLink, null);
+
+        // OperationType.Update is later used for patch method.
+        final Map<String, String> requestHeaders = getRequestHeaders(options, ResourceType.Document, OperationType.Update);
+        Instant serializationStartTimeUTC = Instant.now();
+
+        ByteBuffer content = ByteBuffer.wrap(PatchUtil.serializePatchOperations(patchOperations, options).getBytes(StandardCharsets.UTF_8));
```

Comment:
you could change it so PatchUtil.serializePatchOperations(patchOperations, options) returns byte[], that would help in perf optimization.